### DISCUSSION TITLE
Left bracket. right bracket (bracket mode / unimpaired mode)

### DISCUSSION THREAD

**Discussion by tomhe** [helix-editor/helix#6364] [Q&A] [answered]
I'm on a non-US keyboard where brackets `[` and `]` are typed by two-key combinations `AltGr+8` and  `AltGr+9`, which means that doing e.g. `]t`to move to next function is hard to do.

Is there a way to trigger the bracket mode with other keys on my keyboard that only requires a single keypress?

**Comment by the-mikedavis** [comment, marked as answer]
It isn't currently possible to remap entire sub-keymaps like `g` or `]`/`[`. You can remap [the commands used in that submap](https://github.com/helix-editor/helix/blob/770496511c67cf276f0b0a292e7a82fbd1b5f778/helix-term/src/keymap/default.rs#L104-L129) though:

```toml
# ~/.config/helix/config.toml
[keys.normal.C-t] # rebinds them under 'Ctrl t' but this could be any key
d = "goto_prev_diag"
D = "goto_first_diag"
g = "goto_prev_change"
# ...
